provisioning/hare_setup.py

```diff
@@ -0,0 +1,383 @@
+                    exit(-1)
+
+            if cluster_status_failure == True:
+                exit(-1)
+
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+class Cleanup(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            logging.info('No-op for Cleanup/Reset as of now for Hare ')
+        except Exception as error:
+            logging.error('Failed to clear Hare logs (%s)',
+                          error)
+            exit(-1)
+
+def checkRpm(rpm_name):
+    return os.system('rpm -qa | grep -q ' + rpm_name)
+
+
+def is_cluster_running():
+    if os.system('hctl status >/dev/null') == 0:
+        return True
+    else:
+        return False
+
+
+def bootstrap_cluster():
+    return os.system('hctl bootstrap --mkfs /var/lib/hare/cluster.yaml')
+
+
+def shutdown_cluster():
+    return os.system('hctl shutdown')
+
+
+def list2dict(nodes_data_hctl: list) -> dict:
+    node_info_dict = {}
+    for node in nodes_data_hctl:
+        node_svc_info = {}
+        for service in node['svcs']:
+            node_svc_info[service['name']] = []
+            if (service['status'] == 'started'):
+                node_svc_info[service['name']].append(service['status'])
+        node_info_dict[node['name']] = node_svc_info
+
+    return node_info_dict
+
+
+def check_cluster_status():
+    cluster_desc = None
+    with open("/var/lib/hare/cluster.yaml", 'r') as stream:
+        cluster_desc = yaml.safe_load(stream)
+    cmd = 'hctl status --json'
+    cluster_info = json.loads(execute_command_and_get_output(cmd))
+    nodes_data_hctl = cluster_info['nodes']
+
+    node_info_dict = list2dict(nodes_data_hctl)
+    for node in cluster_desc['nodes']:
+        m0client_s3_cnt = node['m0_clients']['s3']
+        m0client_other_cnt = node['m0_clients']['other']
+        m0ds = node.get('m0_servers', [])
+        for m0d in m0ds:
+            ios_cnt = 0
+            if ('runs_confd' in m0d.keys() and
+                (node_info_dict[node['hostname']]['confd'][0] != 'started')):
+                return -1
+            if m0d['io_disks']['data']:
+                if node_info_dict[node['hostname']]['ioservice'][ios_cnt] != 'started':
+                  return -1
+                ios_cnt += 1
+        if ('s3server' in node_info_dict[node['hostname']].keys() and
```

Comment:
You don't need to check if 's3server' exists in the keys, if s3server count in CDF is 0 then `node_info_dict[node['hostname']]['s3server']` status list will be empty.